tests/countries/test_united_kingdom.py

```diff
@@ -69,32 +70,64 @@ def test_new_years(self):
             "2023-01-02",
         )
         self.assertHolidaysName(f"{name} (Observed)", obs_dt)
+        self.assertNoNonObservedHoliday(obs_dt)
 
-        name = "New Year Holiday [Scotland]"
-        self.assertHolidaysName(
-            name, (f"{year}-01-02" for year in range(1950, 2050))
-        )
-        obs_dt = (
+    def test_new_year_scotland(self):
+        name_new_year = "New Year's Day"
+        name_new_year_holiday = "New Year Holiday"
+        ny_obs_dt = (
             "2000-01-04",
             "2005-01-04",
+            "2006-01-02",
+            "2011-01-04",
+            "2012-01-02",
+            "2017-01-02",
+            "2022-01-04",
+        )
+        nyh_obs_dt = (
+            "2000-01-03",
+            "2005-01-03",
             "2006-01-03",
             "2010-01-04",
-            "2011-01-04",
+            "2011-01-03",
             "2012-01-03",
             "2016-01-04",
             "2017-01-03",
             "2021-01-04",
-            "2022-01-04",
+            "2022-01-03",
         )
-        self.assertHolidaysName(f"{name} (Observed)", obs_dt)
-        self.assertNoNonObservedHoliday(obs_dt)
+
+        for p in {"SCT", "Scotland"}:
```

Comment:
In general one-character names are quite unreadable
```suggestion
        for subdiv in ("SCT", "Scotland"):
```